File: apps/pytest-example/src/example/calculator.py
"""Simple calculator matching the behavior of the Playwright/Vitest examples."""


def add(a: int, b: int) -> int:
    return a + b


def subtract(a: int, b: int) -> int:
    return a - b


def multiply(a: int, b: int) -> int:
    return a * b


def divide(a: int, b: int) -> int:
    if b == 0:
        raise ValueError("Cannot divide by zero")
    return a // b

File: apps/xunit-example/Calculator.cs
namespace XunitExample;

/// <summary>
/// Simple calculator matching the behavior of the Playwright/Vitest examples.
/// </summary>
public static class Calculator
{
    public static int Add(int a, int b) => a + b;
    public static int Subtract(int a, int b) => a - b;
    public static int Multiply(int a, int b) => a * b;

    public static int Divide(int a, int b)
    {
        if (b == 0)
            throw new DivideByZeroException("Cannot divide by zero");
        return a / b;
    }
}
